### ISSUE TITLE
[api-extractor] Result of "export .. from" wrong

### ISSUE THREAD

**Issue by GongT** [microsoft/web-build-tools#1353]
test repo: https://github.com/GongT/test-case-1

## Step 0:
clone and yarn/npm install

## Step 1:
> tsc -p src && api-extractor run --local --verbose

This will generate `./out/dist/xxx.d.ts` with content:
```ts
/** @public */
export declare class A {
	hello();
}

/** @public */
export declare function test(): void;

export declare function test2(): void;

export { }
```

## Step 3:
Copy ts files from src to out, simulate "in place" compile result.
> api-extractor run --local --verbose

This will generate `./out/dist/xxx.d.ts` with **invalid** content: (body is invalid in .d.ts)
```ts
/** @public */
export declare class A {
	hello() {
		console.log('hello world.');
	}
}

/** @public */
export declare function test() {
	console.log('test1');
}

export declare function test2(): void;

export { }
```

# Problem
If I modify "index.d.ts", change all `export ... from './xxx';` into `export ... from './xxx.d';`. `api-extractor run` will produce right result.

And you can see, `test2` have no problem, it's not using `export ... from`.

So I think there is some issue about file resolve.

**Comment by octogonz** [comment]
> Copy ts files from src to out, simulate "in place" compile result.

@GongT Why are you copying source code into your build output folder?

In [Issue #1310](https://github.com/microsoft/web-build-tools/issues/1310) we found that API Extractor gets confused between **Example.ts** and **Example.d.ts** if they appear together in the same folder.   It expects to process the .d.ts file, but the compiler engine recently started following imports into the .ts file extension if present.  Fixing this issue would make these investigations easier.

That said, it's a not a good practice to put **Example.ts** and **Example.d.ts** together in the same folder.  If you fix your toolchain to put source code in the **src** folder, and output in the **out** folder, then the error won't occur.

![DoctorMeme](https://user-images.githubusercontent.com/4673363/60631663-66273e00-9db5-11e9-8e54-13fdf591d8ef.jpg)

**Comment by GongT** [comment]
`tsc` will write output file in place without explicit configure.

Why not support a "default behavior"?

**Comment by GongT** [comment]
Close by duplicate

**Comment by octogonz** [comment]
The fix for #1310 is pretty easy. I agree it's worth fixing, even if it's a recommend usage.
